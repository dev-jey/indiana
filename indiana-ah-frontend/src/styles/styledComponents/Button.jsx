import styled from 'styled-components';

const Button = styled.button`
  background: ${(props) => {
    if (props.danger) {
      return '#CD160B';
    }
    if (props.bgColor) {
      return '#0B41CD';
    }
    return '#fff';
  }}
  font-size: 2rem;
  line-height: 2.7rem;
  align: Center;
  letterspacing: 5%;
  margin: ${props => (props.margin ? props.margin : '')};
  height: ${props => (props.height ? props.height : '')};
  border-radius: 0.4rem;
  text-align: center;
  color: ${(props) => {
    if (props.danger) {
      return '#fff';
    }
    if (props.bgColor) {
      return '#fff';
    }
    return '#0B41CD';
  }}
  cursor: pointer;
  border: ${(props) => {
    if (props.danger) {
      return '0.2rem solid #CD160B';
    }
    if (props.bgColor) {
      return 'none';
    }
    return '0.2rem solid #0B41CD';
  }}
  text-transform: capitalize;
  transition: all 0.3s;
  width: ${props => (props.width ? props.width : 'auto')};
  padding: ${props => (props.sm ? '.4rem 2.5rem' : '.8rem 1.3rem')};
  display: ${props => (props.inlineButton ? 'inline-block' : 'block')};
  :disabled {
    color: #dddddd;
    border: 0.2rem solid #dddddd;
  }
  @media (max-width: 830px) {
    width: auto;
    line-height: 2rem;
    font-size: 1.7rem;
  }
  @media (max-width: 355px) {
    width: auto;
    line-height: 1.5rem;
    padding: ${props => (props.sm ? '.4rem 1.5rem' : '.8rem 1.1rem')};
  }
`;

export default Button;
