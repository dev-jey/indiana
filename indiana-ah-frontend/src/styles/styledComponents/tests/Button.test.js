import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from '../Button.jsx';

describe('testing the button element', () => {
  it('should mount custom button', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background', '#fff');
  });

  it('should apply styles according to passed props', () => {
    const tree = renderer.create(<Button danger inlineButton />).toJSON();
    expect(tree).toHaveStyleRule('color', '#fff');
    expect(tree).toHaveStyleRule(
      'border',
      expect.stringContaining('0.2rem solid #CD160B')
    );
    expect(tree).toHaveStyleRule('display', expect.stringContaining('inline-block'));
  });
});
