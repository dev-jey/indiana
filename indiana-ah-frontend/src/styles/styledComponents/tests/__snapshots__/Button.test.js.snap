// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the button element should mount custom button 1`] = `
.c0 {
  background: #fff;
  font-size: 2rem;
  line-height: 2.7rem;
  align: Center;
  -webkit-letterspacing: 5%;
  -moz-letterspacing: 5%;
  -ms-letterspacing: 5%;
  letterspacing: 5%;
  border-radius: 0.4rem;
  text-align: center;
  color: #0B41CD;
  cursor: pointer;
  border: 0.2rem solid #0B41CD;
  text-transform: capitalize;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: auto;
  padding: .8rem 1.3rem;
  display: block;
}

.c0:disabled {
  color: #dddddd;
  border: 0.2rem solid #dddddd;
}

@media (max-width:830px) {
  .c0 {
    width: auto;
    line-height: 2rem;
    font-size: 1.7rem;
  }
}

@media (max-width:355px) {
  .c0 {
    width: auto;
    line-height: 1.5rem;
    padding: .8rem 1.1rem;
  }
}

<button
  className="c0"
/>
`;
